aptos-move/aptos-intent/src/codegen.rs

```diff
@@ -0,0 +1,489 @@
+// Copyright © Aptos Foundation
```

Comment:
Wondering if we should move this into `move-binary-format`. A builder library would have a lot of use cases besides the scope of the intent engine

Reply:
I think parts of the code can potentially go to move-binary-format? But this would require some refactoring to be able to handle both module and script so I would avoid doing that in this PR.